### ISSUE TITLE
Default Stack and Templates 

### ISSUE THREAD

**Issue by sammachin** [flowforge/flowforge#989]
### Description

Make the 1st Stack and Template on the new project screen selected be default so a user doesn't have to make a choice.

This is in keeping with our priniciple of Sensible Defaults

**Comment by sammachin** [comment]
Where there is only a single option available to a user they should not even be shown the choice it should just be set automatically

**Comment by knolleary** [comment]
We already have the concept of a ProjectType's default stack - and it gets preselected on the Create Project page when that ProjectType is selected.

However there is a bug. If a stack has been replaced by another, the ProjectType's default stack entry isn't being updated - so still points to the old stack. As that stack doesn't show up on the create project page, nothing gets selected.

I've fixed up the page so that it ensures *something* is selected. I'll also sort out the bug described above.

As for hiding the selection entirely - I think there is a case to be made to hide the Template option if there's only one - as we don't yet have much functionality for end users around multiple templates.

But I do think stacks are different because they determine details of the project that are useful to know up front - NR version for example (and will be better once we have #915).

So I propose to hide the Template row if there's only one, but leave the Stack there.
